### PR TITLE
Added tcpWER scoring

### PR DESCRIPTION
**Pull request by popcornell**

### CODE REVIEW

File: scripts/chime8/pipeline/local/eval/run_eval.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# script adapted from https://github.com/espnet/espnet/blob/master/egs2/chime7_task1/asr1/local/da_wer_scoring.py
+
+import glob
+import json
+from pathlib import Path
+import os
+
+from omegaconf import DictConfig, OmegaConf, open_dict
```
[the remark this answers is not in the document]
Comment (tango4j) [reply scripts/chime8/pipeline/local/eval/run_eval.py]: removed

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 2]
```diff
@@ -697,6 +694,123 @@
...
+        torch.cuda.empty_cache()
+    else:
+        gc.collect()
+
+    del ms_embs_scaled_vadmasked, ms_silsp_embs, selected_ss_mc_embs, ms_ts_scaled, vad_decision_scaled, vad_decision_base
+    if get_rttm_with_the_finest_scale: 
+        timestamps = time_stamps[-1][:max_scm][cluster_labels_infer != -1]/feat_per_sec
+        cluster_labels = cluster_labels_infer[cluster_labels_infer != -1].cpu().numpy()
+    else:
+        timestamps = ms_ts_scaled[vad_decision_scaled, :]
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/parts/utils/speaker_utils.py:790]: Changed the order of `del`

[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/parts/utils/speaker_utils.py:790]: Changed the order of del

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 3]
```diff
@@ -697,6 +694,123 @@
...
+        lines_cluster_labels.extend([f'{uniq_id} {seg_line}\n' for seg_line in lines])
+    hypothesis = labels_to_pyannote_object(labels, uniq_name=uniq_id)
+    hyp_entry = [uniq_id, hypothesis]
+    rttm_file = audio_rttm_values.get('rttm_filepath', None)
+    if rttm_file is not None and os.path.exists(rttm_file) and not no_references:
+        ref_labels = rttm_to_labels(rttm_file)
+        reference = labels_to_pyannote_object(ref_labels, uniq_name=uniq_id)
+        ref_entry = [uniq_id, reference]
+    else:
+        no_references = True
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/parts/utils/speaker_utils.py]: Removed

File: nemo/collections/asr/models/msdd_v2_models.py [hunk 4]
```diff
@@ -681,9 +693,8 @@
         self._validation_dl = self.__setup_dataloader_from_config(config=val_data_config,)
         return max_len_ms_ts, max_len_scale_map 
 
-    def setup_multiple_test_data(self, test_data_config):
-        """
-        MSDD does not use multiple_test_data template. This function is a placeholder for preventing error.
+    def setup_multiple_test_data(self):
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/models/msdd_v2_models.py]: removed

File: nemo/collections/asr/models/msdd_v2_models.py [hunk 5]
```diff
@@ -610,8 +621,8 @@
             shift = ovl_len 
         else:
             shift = self.cfg.session_len_sec - 2 * ovl_len
-            
-        session_durations_list = self.setup_test_manifest_list(test_manifest_jsons, shift, ovl_len)
+
+        session_durations_list = self.setup_test_manifest_list(test_manifest_jsons, shift, ovl_len, white_uniq_id=uniq_id)
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/models/msdd_v2_models.py]: removed

**Comment by tango4j** [comment]
Made some changes to resolve codeQL issues, merged main then tested.
Now it is ready to merge to main.

**Comment by popcornell** [comment]
Ok thanks ! Merging then

**Pull request merged**